### PR TITLE
Feature/objc wrapper ii

### PR DESCRIPTION
**Pull request by drhaynes**
Adds rudimentary search error handling for obj-c. Can be augmented with what goes into OSAPIResponse at a later date.

### CODE REVIEW

File: Search/OSPlacesSearchService.swift [hunk 1]
```diff
@@ -39,3 +39,7 @@ import Fetch
     }
 
 }
+
+private func nsErrorFromSearchError(error: SearchError) -> NSError {
+    return NSError(domain: OSSearchErrorDomain, code: error.rawValue(), userInfo: nil)
```
Comment (dhardiman) [inline Search/OSPlacesSearchService.swift]: `NSLocalizedDescriptionKey` might be sensible here? Not sure I'd expect canned responses to be in the API response framework.

File: Search/OSSearchError.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+//  Created by David Haynes on 16/03/2016.
+//  Copyright © 2016 Ordnance Survey. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+/**
+ *  Error domain for use with the search framework
+ */
+NSString *const OSSearchErrorDomain = @"uk.os.search.error";
```
Comment (dhardiman) [inline Search/OSSearchError.h]: The actual string should be in a .m file. This should just be `extern NSString *const OSSearchErrorDomain;`

File: Search/OSPlacesSearchService.swift (hunk 1 above)
Comment (drhaynes) [inline Search/OSPlacesSearchService.swift]: Yeah, wasn't sure what we would include. Will add them.

**Comment by drhaynes** [comment]
Updated

**Comment by dhardiman** [comment]
I'll look at some of these to port in to OSAPIResponse this morning.

**Pull request merged**